### PR TITLE
Updated ILGPU project to v0.9.0-beta1 release.

### PR DESCRIPTION
**Pull request by MoFtZ**
Depends on #34 and https://github.com/m4rs-mt/ILGPU/pull/206.

ISSUE: CuBlas does not support `long Length`.

ISSUE: Histogram and RadixSort need to be updated to support `long Length`. (FIXED)

ISSUE: RadixSort cannot access the raw bits of `Half`. (FIXED)

### CODE REVIEW

**Comment by m4rs-mt** [comment]
> ISSUE: RadixSort cannot access the raw bits of Half.

I will make the property `RawValue` public such that we can use it to extract the underlying raw bits.

**Comment by m4rs-mt** [comment]
@MoFtZ Forget what I just posted 😞 This won't work in the end because this feature is not available on all architectures and requires additional intrinsic operations (at least for OpenCL). In PTX we can simply reuse the same register (`rs XXX`).

**Comment by m4rs-mt** [comment]
I would propose that we add additional `FloatAsInt` and `IntAsFloat` overloads to the `Interop` type that can perform this raw-value/half-value conversions.

**Comment by MoFtZ** [comment]
It's already there 🤣

**Comment by m4rs-mt** [comment]
@MoFtZ I've seen it 😄 However, we need to double-check whether they actually work as intended 🚀

**Comment by m4rs-mt** [comment]
There is a minor issue in the PTX backend. I am going to prepare a patch 🤓

**Comment by MoFtZ** [comment]
> ISSUE: Histogram and RadixSort need to be updated to support `long Length`.
> ISSUE: RadixSort cannot access the raw bits of `Half`.

Fixed Histogram and RadixSort implementations for 64-bit allocations and Half type.

Depends on https://github.com/m4rs-mt/ILGPU/pull/209.

**Comment by MoFtZ** [comment]
@m4rs-mt This is going to require a bigger change.

For example, `GridExtensions.ComputeGridStrideLoopExtent` accepts `Index1 numDataElements` as a parameter. We typically call the method using `arrayView.Length` which returns `long`. The issue is that the implicit cast from `long` to `Index1` is hiding potential overflows at runtime.

Also, if an algorithm is supplied an `ArrayView` that is of 32-bit length, should it use an specialized kernel for 32-bit lengths rather than 64-bit lengths?

**Comment by m4rs-mt** [comment]
@MoFtZ You are right. Maybe we have to provide specialized versions that even take the limitations of the current thread-grid sizes (32-bits for Cuda and OpenCL) into account.

One option would be to create additional kernel launcher overloads (in the scope of the ILGPU project) that circumvent these driver limitations by breaking 64-bit thread grids down to multiple 32-bit kernel launches.

File: Src/ILGPU.Algorithms/Resources/ErrorMessages.resx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="NotSupportedArrayView64" xml:space="preserve">
+    <value>This function does not currently support ArrayViews of length &gt; int.MaxValue.</value>
```
Comment (MoFtZ) [inline Src/ILGPU.Algorithms/Resources/ErrorMessages.resx:121]: NOTE: Needs to be reviewed for correct wording.

`This function does not currently support ArrayViews of length > int.MaxValue.`

**Comment by MoFtZ** [comment]
ISSUE: `SequencerHalf` unit test for length over 2048 is failing.

Looks like Cuda accuracy for Half type is slightly off.
```
CPU
(Half)2202 = 2202
(Half)2203 = 2202
(Half)2204 = 2204
(Half)2205 = 2204
(Half)2206 = 2206
(Half)2207 = 2206

CUDA
(Half)2202 = 2202
(Half)2203 = 2204
(Half)2204 = 2204
(Half)2205 = 2204
(Half)2206 = 2206
(Half)2207 = 2208
```

**Review by m4rs-mt — APPROVED** [review]
Awesome work 🚀

**Pull request merged**